src/openforms/config/models.py

```diff
@@ -661,14 +661,25 @@ def as_dict(self):
 class CSPSettingManager(models.Manager.from_queryset(CSPSettingQuerySet)):
     @transaction.atomic
     def set_for(
-        self, obj: models.Model, settings: list[tuple[CSPDirective, str]]
+        self,
+        obj: models.Model,
+        settings: list[CSPEntry],
     ) -> None:
         """
-        Deletes all the connected csp settings and creates new ones based on the new provided data.
+        Deletes all the connected CSP settings and creates new ones based on the new provided data.
+
+        :param obj: The configuration model providing this CSP entry.
+        :param `*settings`: Either a two-tuple or a three-tuple containing values used to create
+          the underlying ``CSPSetting`` model.
```

Comment:
So I first went from a list of tuples to a dto dataclass because I've added this extra `identifier` attribute. But I've then moved `identifier` to the signature of `set_for`.

I'm not a fan of DTO objects in general, so we can either revert to the old way (list of tuples), or use unsaved instances instead. But having unsaved instances feels weird as well:

```python
CSPSetting.objects.set_for(settings=[CSPSetting(...), CSPSetting(...)])
```

What do you think?

Reply:
List of tuples is probably best then as it avoids potentially dealing with circular imports of the model.